packages/integration-tests/suites/replay/sessionExpiry/test.ts

```diff
@@ -0,0 +1,95 @@
+import { expect } from '@playwright/test';
+import type { ReplayEvent } from '@sentry/types';
+
+import { sentryTest } from '../../../utils/fixtures';
+import { envelopeRequestParser } from '../../../utils/helpers';
+import { getExpectedReplayEvent } from '../../../utils/replayEventTemplates';
+import {
+  getFullRecordingSnapshots,
+  getIncrementalRecordingSnapshots,
+  getReplaySnapshot,
+  normalize,
+  shouldSkipReplayTest,
+  waitForReplayRequest,
+} from '../../../utils/replayHelpers';
+
+// Session should expire after 2s - keep in sync with init.js
+const SESSION_TIMEOUT = 2000;
+
+sentryTest('handles an expired session', async ({ getLocalTestPath, page }) => {
+  if (shouldSkipReplayTest()) {
+    sentryTest.skip();
+  }
+
+  const reqPromise0 = waitForReplayRequest(page, 0);
+  const reqPromise1 = waitForReplayRequest(page, 1);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+
+  const replayEvent0 = envelopeRequestParser(await reqPromise0) as ReplayEvent;
```

Comment:
super-l: I initially wanted to write we should use `getReplayEvent` here but then I realized that it essentially just wraps `envelopeRequestParser`. So overall my logaf is super low here other than that it's an inconsistency between tests. Feel free to ignore :)

Reply:
ah, good point, will update this everywhere!